### PR TITLE
Convert the build scripts from Ant to Gradle.

### PR DESCRIPTION
**Pull request by smillst**
I tried to make the Gradle tasks do the same thing as the Ant tasks did.  However, since we are ending support for JSR308, I removed all references to it in the build scripts except for the jtreg tests, which is explained in a comment in that task.  CheckerMain still uses it though because plume-lib still uses annotations in comments and because we haven't removed mentions of annotations in comments from the manual.

See [developer-manual.html](https://github.com/typetools/checker-framework/pull/1871/files#diff-521ff72506c5df28427c8b58da6cb293) for overview of the Gradle tasks.

The release scripts need to be updated for this change, but that will happen after this is merged.

After this is merged, we will remove the ant build scripts.  After that, we should relocate source and tests files so that each source set is in its own directory.

### CODE REVIEW

File: .travis-build.sh [hunk 1]
```diff
@@ -88,14 +72,15 @@ if [[ "${GROUP}" == "downstream" || "${GROUP}" == "all" ]]; then
   else
     CFISLUGOWNER=${SLUGOWNER}
   fi
+  CFISLUGOWNER=smillst
   set -e
   echo "Running:  (cd .. && git clone --depth 1 https://github.com/${CFISLUGOWNER}/checker-framework-inference.git)"
-  (cd .. && git clone --depth 1 https://github.com/${CFISLUGOWNER}/checker-framework-inference.git) || (cd .. && git clone --depth 1 https://github.com/${CFISLUGOWNER}/checker-framework-inference.git)
+  (cd .. && git clone --depth 1 -b gradle https://github.com/${CFISLUGOWNER}/checker-framework-inference.git) || (cd .. && git clone -b gradle --depth 1 https://github.com/${CFISLUGOWNER}/checker-framework-inference.git)
```
Comment (smillst) [inline .travis-build.sh]: Note: This needs to be reverted before merge.  (See PR https://github.com/typetools/checker-framework-inference/pull/71)

File: framework/tests/src/tests/RangeTest.java [hunk 2]
```diff
@@ -183,6 +183,7 @@ public void remove() {
 
         public ValuesInRangeIterator(Range range) {
             this.range = range;
+            Range.IGNORE_OVERFLOW = false;
```
Comment (mernst) [inline framework/tests/src/tests/RangeTest.java]: How is this change related to the rest of the pull request?

File: .travis-build.sh [hunk 3]
```diff
@@ -88,14 +72,15 @@ if [[ "${GROUP}" == "downstream" || "${GROUP}" == "all" ]]; then
   else
     CFISLUGOWNER=${SLUGOWNER}
   fi
+  CFISLUGOWNER=smillst
```
Comment (mernst) [inline .travis-build.sh]: This should be changed before being merged.

Comment (smillst) [reply .travis-build.sh]: Yes, it's related to the change below.

**Comment by smillst** [comment]
> There are still occurrences of -target 7:

This pull request doesn't disable annotations in comments, so these are fine.  (A separate pull request will do that along with updating the manual.)

File: framework/tests/src/tests/RangeTest.java (hunk 2 above)
Comment (smillst) [reply framework/tests/src/tests/RangeTest.java]: This pull request uncovered a bug in the Value Checker.  I'll move the fix to another branch.

File: framework/tests/annotationclassloader/Makefile [hunk 4]
```diff
@@ -47,34 +45,35 @@ demo1:
 demo2:
 	@echo "***** This command is expected to produce an error on line 7:"
 	$(JAVAC) \
-	  -J-Xbootclasspath/p:$(FRAMEWORKJAR):$(STUBPARSERJAR):$(JAVACJAR) \
-	  -classpath :$(PROJECTDIR) \
+	  -processorpath $(FRAMEWORKJAR) \
```
Comment (wmdietl) [inline framework/tests/annotationclassloader/Makefile]: Are these "fat" jars or do they just contain the individual project files? It seems a bit odd that the different demos in this file use different `processorpath`s, even though they probably all have the same dependencies. It would probably be best if all of them depend on the `build` versions and you add a new variable for the union of all of them.

File: framework/tests/src/tests/RangeTest.java (hunk 2 above)
Comment (wmdietl) [reply framework/tests/src/tests/RangeTest.java]: Is this something that should be pulled into this PR?

File: framework/tests/src/tests/WholeProgramInferenceValidationTest.java [hunk 5]
```diff
@@ -19,6 +19,13 @@ public WholeProgramInferenceValidationTest(List<File> testFiles) {
         super(testFiles, WholeProgramInferenceTestChecker.class, "value", "-Anomsgtext");
     }
 
+    @Override
+    public void run() {
+        if (new File("tests/whole-program-inference/annotated/").exists()) {
+            super.run();
+        }
```
Comment (wmdietl) [inline framework/tests/src/tests/WholeProgramInferenceValidationTest.java]: Should this raise a user-friendly error if the directory is not found?

File: .travis-build.sh (hunk 1 above)
Comment (wmdietl) [reply .travis-build.sh]: I merged that PR, so the change above can be undone.

File: framework/tests/annotationclassloader/Makefile (hunk 4 above)
Comment (smillst) [reply framework/tests/annotationclassloader/Makefile]: They are fat jars.  The tests are suppose to use different processorpaths to test loading annotations from a jar file and loading annotations from classes.
